File: tests/perf2/test-hello-world.js
/*
 *  Obligatory test.
 */

if (typeof window == 'undefined') window = {}
if (!window.performance) window.performance = {}
window.performance.now = window.performance.now || Date.now

if (typeof print != 'undefined') log = print
else log = console.log

function test () {
  var start
  var duration

  start = window.performance.now()

  // begin test
  log('Hello, world!')
  // end test

  duration = window.performance.now() - start

  log(duration)
}

test()

File: tests/perf2/test-hello-world.c
/*
 * Obligatory test.
 */

#include "stdio.h"
#include "time.h"

int main(int argc, char **argv) {
     int msec;
     clock_t start, duration;

     start = clock();

     // begin test
     printf("Hello, world!\n");
     // end test

     duration = clock() - start;
     msec = duration * 1000 / CLOCKS_PER_SEC;

     printf("%d\n", msec);

     return 0;
}
